microsoft-edge/webdriver-chromium/ie-mode.md

```diff
@@ -0,0 +1,222 @@
+
+<a id="drive-ie-mode-entire-sample"></a>
+
+<!-- todo -->
+Insert Python sample here
+
+* * *
+
+The following sections explain the steps in this sample in more detail.
+
+
+<!-- ====================================================================== -->
+## Create an InternetExplorerDriverService
+
+### [C#](#tab/c-sharp/)
+
+<a id="drive-ie-mode-ieService"></a>
+
+To create an `InternetExplorerDriverService`, use the `CreateDefaultService` method.  Pass in the following parameters:
+
+*  `dir`, which is the directory where you downloaded IEDriver to.
+*  `driverexe`, which is the name of the IEDriver executable.
+
+```csharp
+//change the path accordingly
+var dir = @"C:\<path to IEDriver download>";
+var driverexe = "IEDriverServer.exe";
+
+var ieService = InternetExplorerDriverService.CreateDefaultService(dir, driverexe);
+```
+
+### [Python](#tab/python/)
+
+<a id="drive-ie-mode-ieService"></a>
+
+<!-- todo -->
+Insert Python sample here
+
+* * *
+
+
+<!-- ====================================================================== -->
+## Define InternetExplorerOptions with additional capabilities for Microsoft Edge
+
+### [C#](#tab/c-sharp/)
+
+<a id="drive-ie-mode-ieOptions"></a>
+
+1. Define a new variable, `ieOptions`, by instantiating `InternetExplorerOptions` with the `IgnoreZoomLevel` property set to `true`.
+
+1. Call the `AddAdditionalCapability` method, with the `ie.edgechromium` property set to `true`, and `ie.edgepath` set to the path of the Microsoft Edge executable:
+
+```csharp
+var ieOptions = new InternetExplorerOptions { IgnoreZoomLevel = true };
+ieOptions.AddAdditionalCapability("ie.edgechromium", true);
+//change the path accordingly
+ieOptions.AddAdditionalCapability("ie.edgepath", 
+       "C:/Program Files (x86)/Microsoft/Edge/Application/msedge.exe");
+```
+
+### [Python](#tab/python/)
+
+<a id="drive-ie-mode-ieOptions"></a>
+
+<!-- todo -->
+Insert Python sample here
+
+* * *
+
+
+<!-- ====================================================================== -->
+## Start InternetExplorerDriver
+
+### [C#](#tab/c-sharp/)
+
+<a id="drive-ie-mode-ieDriver"></a>
+
+Finally, start `InternetExplorerDriver` and pass it the previously defined `ieService` and `ieOptions`.  IEDriver launches Microsoft Edge in IE mode.  All page navigation and subsequent interactions occur in IE mode.
+
+```csharp
+InternetExplorerDriver reqDriver = new InternetExplorerDriver(ieService, ieOptions);
+```
+
+### [Python](#tab/python/)
+
+<a id="drive-ie-mode-ieDriver"></a>
+
+<!-- todo -->
+Insert Python sample here
+
+* * *
+
+
+<!-- ====================================================================== -->
+## Known limitations
+
+When automating IE mode in Microsoft Edge, you must add a short wait between opening a new window with [window.open]https://developer.mozilla.org/docs/Web/API/Window/open) and getting handles to the new window with the [Get Window Handles](https://www.w3.org/TR/webdriver2/#get-window-handles) command.
```

Comment:
This isn't a requirement. We do the waiting internally. The limitation is that if multiple windows are opened at nearly the same time, then our wait logic we added to the "new window" command might not be able to accurately detect which window is the one you opened.

Some potential workarounds are:
- Space out your new window commands and avoid page-initiated popups etc so the driver doesn't get confused.
- Set the page load strategy to "none" and poll the window handles yourself since you (i.e. the developer) should hopefully know better than IEDriver which window is the one you are looking for.

So the workarounds look a lot like what you've described here. I would just reframe it a bit that you don't need to resort to these workarounds unless your existing test code is having trouble with new window commands.